### PR TITLE
Adds Legend Widget for both Ordinal and Rainbow

### LINKED ISSUE #523: Add "Ordinal" Legend Widget
Add a legend widget which "just works" with any of our chart APIs

I think it should be derived from other/Table with CSS styling to hide the borders.  This will allow automatic paging or scrolling as users see fit.

### LINKED ISSUE #524: Add "Rainbow" Legend Widget
This one will need some thought, but it will be used by the choropleths so should display the "rainbow" of the palette like you see on test.html - then on the RHS we would display the entities with little arrows pointing to their colours?

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #523
Fixes #524
Fixes #578

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage Can you add some test links - and your commit message needs the word fixes infront of the issue numbers.

File: src/other/Legend.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "./Table", "../chart/MultiChart", "../layout/Grid", "css!./Legend"], factory);
+    } else {
+        root.other_Table = factory(root.d3, root.other_Table, root.chart_MultiChart, root.layout_Grid);
```
Comment (GordonSmith) [inline src/other/Legend.js]: This is 1 reason why your travis is failing

**Comment by jbrundage** [comment]
Dermatology:
https://rawgit.com/jbrundage/Visualization/OrdinalLegend-1.6.0/demos/dermatology.html?src/other/Legend

Legend test page:
https://rawgit.com/jbrundage/Visualization/OrdinalLegend-1.6.0/apps/SampleSite/legend.html?src/map/ChoroplethStates

**Comment by GordonSmith** [comment]
Have you tried putting it in the MegaChart yet?

File: src/other/Legend.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            else if (paletteType === "rainbow"){
+                var colorArr = widget._palette.colors().reverse();
+                var steps = colorArr.length;
+                var weightMin = widget._dataMinWeight;
+                var weightMax = widget._dataMaxWeight;
+                for(var x = 0;x<steps;x++){
+                    var stepWeightDiff = parseInt((weightMax - weightMin) / steps);
+                    var lower,higher;
+                    if(x === 0){
+                        //lower = commaSeparateNumber(weightMin + (stepWeightDiff*x));
```
Comment (GordonSmith) [inline src/other/Legend.js]: remove comments

File: src/other/Legend.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                        higher = commaSeparateNumber(weightMin + stepWeightDiff*(x+1));
+                        dataArr.push([_htmlColorBlock(colorArr[x]),"0 - " + higher]);
+                    } else if (x+1 === steps){
+                        lower = commaSeparateNumber(weightMin + (stepWeightDiff*x) + 1);
+                        //higher = commaSeparateNumber(weightMin + stepWeightDiff*(x+1));
```
Comment (GordonSmith) [inline src/other/Legend.js]: remove comments

**Comment by jbrundage** [comment]
I haven't tried putting it in the MegaChart yet.

**Comment by GordonSmith** [comment]
@jbrundage - we need to come up with a strategy for including this in all charts - Mega Chart is one - but also we could use layers to put it on top of chart with a opacity that goes to 1 when mouse over happens (and a button in the surface maybe - we need ideas!!!

**Pull request merged**